modules/tide_oauth/README.md

```diff
@@ -0,0 +1,50 @@
+## Tide OAuth
+
+### Prerequisites
+1. Consumers module: https://www.drupal.org/project/consumers
+2. Simple OAuth module: https://www.drupal.org/project/simple_oauth
+
+### Installation
+1. _(Optional)_ Generate a key pair and set environment variables
+    ```shell script
+    $ openssl genrsa -out /tmp/private.key 4096
+    $ openssl rsa -in /tmp/private.key -pubout > /tmp/public.key
+    $ export TIDE_OAUTH_PRIVATE_KEY=`cat /tmp/private.key`
+    $ export TIDE_OAUTH_PUBLIC_KEY=`cat /tmp/public.key`
+    ```
+2. Enable the `tide_oauth` module.
+    * If the TIDE_OAUTH_ environment variables are set, the module will copy
+    the keys to `private://oauth.key` and `private://oauth.pub`.
+    * Otherwise, the module will generate a new key pair.
+3. _(Optional - **Lagoon only**)_ Add a `post-rollout` task to generate the OAuth
```

Comment:
Is there any reason this couldn't be an openssl command to generate on the fly from scratch?

Reply:
if the key changes, all existing tokens become invalid, even if they are not expired.

Reply:
So if the deploy step checked if the key existed first before trying to create it, then it would work?

Reply:
@anthony-malkoun yes https://github.com/dpc-sdp/tide_core/pull/172/files#diff-6f2392828bd5ca4ff53608718b20927df9c8d6d10a19951924aa36803885473dR51

Reply:
it only happens upon installation. If it's already on Prod and you launch a new dev env, the key pair doesn't exist on the new env and it still fails. There is no mechanism for Drupal to check the key pair upon init, hence the `poll-rollout` task.

Reply:
Do you see any issue if we use a HMAC key rather than an RSA key @sonnykt ? It should just be the instructions that change. HMAC will be a single line token that can then be defined at the project level in Lagoon (and overridden for production) and will be able to be managed by our configuration management.